packages/core/src/utilities/VoxelValue.ts

```diff
@@ -0,0 +1,235 @@
+      this.modifiedSlices.add(pointIJK[2]);
+      VoxelValue.addBounds(this.boundsIJK, pointIJK);
+    }
+  };
+
+  public toIJK(index: number): Point3 {
+    return [
+      index % this.width,
+      Math.floor((index % this.frameSize) / this.width),
+      Math.floor(index / this.frameSize),
+    ];
+  }
+
+  public toIndex(ijk: Point3) {
+    return ijk[0] + ijk[1] * this.width + ijk[2] * this.frameSize;
+  }
+
+  public getBoundsIJK(): BoundsIJK {
+    if (this.boundsIJK[0][0] < this.dimensions[0]) {
+      return this.boundsIJK;
+    }
+    return this.dimensions.map((dimension) => [0, dimension - 1]) as BoundsIJK;
+  }
+
+  public forEach = (callback, options?) => {
+    const boundsIJK = options?.boundsIJK || this.getBoundsIJK();
+    const { isWithinObject } = options || {};
+    if (this.map) {
+      // Optimize this for only values in the map
+      for (const index of this.map.keys()) {
+        const pointIJK = this.toIJK(index);
+        const value = this._get(index);
+        const callbackArguments = { value, index, pointIJK };
+        if (isWithinObject?.(callbackArguments) === false) {
+          continue;
+        }
+        callback(callbackArguments);
+      }
+    } else {
+      for (let k = boundsIJK[2][0]; k <= boundsIJK[2][1]; k++) {
+        const kIndex = k * this.frameSize;
+        for (let j = boundsIJK[1][0]; j <= boundsIJK[1][1]; j++) {
+          const jIndex = kIndex + j * this.width;
+          for (
+            let i = boundsIJK[0][0], index = jIndex + i;
+            i <= boundsIJK[0][1];
+            i++, index++
+          ) {
+            const value = this.getIndex(index);
+            const callbackArguments = { value, index, pointIJK: [i, j, k] };
+            if (isWithinObject?.(callbackArguments) === false) {
+              continue;
+            }
+            callback(callbackArguments);
+          }
+        }
+      }
+    }
+  };
+
+  public static addBounds(bounds: BoundsIJK, point: Point3) {
+    bounds.forEach((bound, index) => {
+      bound[0] = Math.min(point[index], bound[0]);
+      bound[1] = Math.max(point[index], bound[1]);
+    });
+  }
+
+  /**
+   *  Creates a volume value accessor
+   */
+  public static volumeVoxelValue(
+    dimensions: Point3,
+    scalarData
+  ): VoxelValue<number> {
+    const voxels = new VoxelValue(
+      dimensions,
+      (index) => scalarData[index],
+      (index, v) => {
+        const isChanged = scalarData[index] !== v;
+        scalarData[index] = v;
+        return isChanged;
+      }
+    );
+    voxels.scalarData = scalarData;
+    return voxels;
+  }
+
+  /**
+   * Creates a volume map value accessor
+   */
+  public static mapVoxelValue<T>(dimension: Point3): VoxelValue<T> {
+    const map = new Map<number, T>();
+    const voxelValue = new VoxelValue(
+      dimension,
+      map.get.bind(map),
+      (index, v) => map.set(index, v) && true
+    );
+    voxelValue.map = map;
+    return voxelValue;
+  }
```

Comment:
I'm not sure about these two names either but can't think of good ones either. For instance when i was reading the getStrategyData code 

`segmentationVoxelValue: VoxelValue.volumeVoxelValue(
      dimensions,
      segmentationScalarData
    ),`

I absolutely did not expect this `new` logic